src/main/java/cd/go/contrib/elasticagents/docker/executors/JobCompletionRequestExecutor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cd.go.contrib.elasticagents.docker.executors;
+
+import cd.go.contrib.elasticagents.docker.*;
+import cd.go.contrib.elasticagents.docker.requests.JobCompletionRequest;
+import com.thoughtworks.go.plugin.api.response.DefaultGoPluginApiResponse;
+import com.thoughtworks.go.plugin.api.response.GoPluginApiResponse;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static cd.go.contrib.elasticagents.docker.DockerPlugin.LOG;
+import static java.text.MessageFormat.format;
+
+public class JobCompletionRequestExecutor implements RequestExecutor {
+    private final JobCompletionRequest jobCompletionRequest;
+    private final AgentInstances<DockerContainer> agentInstances;
+    private final PluginRequest pluginRequest;
+
+    public JobCompletionRequestExecutor(JobCompletionRequest jobCompletionRequest, AgentInstances<DockerContainer> agentInstances, PluginRequest pluginRequest) {
+        this.jobCompletionRequest = jobCompletionRequest;
+        this.agentInstances = agentInstances;
+        this.pluginRequest = pluginRequest;
+    }
+
+    @Override
+    public GoPluginApiResponse execute() throws Exception {
+        PluginSettings pluginSettings = pluginRequest.getPluginSettings();
+        String elasticAgentId = jobCompletionRequest.getElasticAgentId();
+        Agent agent = new Agent();
+        agent.setElasticAgentId(elasticAgentId);
+        LOG.debug(format("[Job Completion] Terminating elastic agent with id {0} on job completion {1}.", agent.elasticAgentId(), jobCompletionRequest.jobIdentifier()));
```

Comment:
You do not need to format the log message, you should be able to use the slf4j parameterized messages, 
e.g `LOG.debug("[Job Completion] Terminating elastic agent with id {} on job completion {}.", agent.elasticAgentId(), jobCompletionRequest.jobIdentifier());`

Reply:
Any performance impact with this style? @maheshp

Reply:
> Any performance impact with this style?

https://www.slf4j.org/faq.html#logging_performance

Reply:
Agreed that Templating is better than String concatenation. Would like to follow the same convention followed for string templatization across the project. Found this style [java.text.MessageFormat.format] used in multiple other places as well. Hence used it here too.

Reply:
String templating is a relic of the past. We'd earlier used log4j as the backend for logging, which is why we used string templating instead of concatenation. Sometimes coupled with a `if(log.isDebugEnabled())` wrapper.

If you see any occurances of `format` in a log statement. Feel free to change it to use sl4fj style templating.

Reply:
Btw the logger used in the plugin is a custom logger which is a wrapper class
com.thoughtworks.go.plugin.api.logging.Logger @ketan @maheshp . sl4j logger has its own formatting support.